cudaungappedextender/include/claraparabricks/genomeworks/cudaungappedextender/cudaungappedextender.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#pragma once
+
+namespace claraparabricks
+{
+namespace genomeworks
+{
+namespace cudaungappedextender
+{
+
+struct SeedPair
+{
+    /// position of first sketch element in query_read_id_
+    position_in_read_t query_position_in_read;
+    /// position of second sketch element in target_read_id_
+    position_in_read_t target_position_in_read;
+};
+
+struct ScoredSegmentPair
+{
+    /// Seed for the segment
+    Seed seed_pair;
+    /// length of the segment
+    int32_t length;
+    /// score of the segment
+    int32_t score;
+};
+
+/// CUDA Ungapped Extension object
+class UngappedExtender
+{
+public:
+    /// \brief Constructor Prototype
+    UngappedExtender(int* h_sub_mat, int input_xdrop, bool input_noentropy,
```

Comment:
Should I move the xdrop parameter to the extend_aync call or is it a one-time set-and-forget?

Reply:
It is one-time set-and-forget.